oligos/models.py

```diff
@@ -34,13 +43,354 @@ class Oligo(models.Model):
+    return newseq
+
+def cleanpos(pos):
+    if 'antisense' in pos:
+        return 'A'
+    if 'sense' in pos:
+        return 'S'
+    return 'U'
+
+def getusages(usage):
+    v = usage.split('\n')
+    ulist=[]
+    otherfound = False
+    for use in v:
+        use.strip()
+        if use=='':
+            pass
+        elif Usage.objects.filter(usage__iexact=use):
+            u=Usage.objects.get(usage__iexact=use)
+            ulist.append(u)
+        else:
+            if not otherfound:
+                u=Usage.objects.get(usage='Other')
+                ulist.append(u)
+                otherfound=True
+    return ulist;
+
+def unicodeline(line):
+    line2=u''
+    for c in line:
+        try:
+            c=unicode(c,'utf-8')
+        except:
+            c='\n'
+        line2+=c
+    return line2
+
+def parseoligo(line):
+    v=line.split('","')
+    dic={}
+    dic['oligoid']=v[0].strip('"')
+    dic['olduser']=v[1]
+    dic['info']=v[2]
+    dic['cdate']=v[3]
+    dic['mdate']=v[4]
+    dic['sequence']=v[5]
+    dic['reference']=v[6]
+    dic['gene_locus']=v[7]
+    dic['organism']=v[8]
+    dic['usage']=v[9]
+    dic['primer_position']=v[10]
+    dic['primer_partner']=v[11]
+    dic['company']=v[12]
+    dic['concentration']=v[13]
+    dic['grade']=v[14].strip('"')
+    for k,v in dic.iteritems():
+        v=v.replace('\x0b','\n')
+        v=v.strip()
+        dic[k]=v
+    return dic
+
+def writeerrorstring(dic,message):
+    f2 = open('c:/pydata/boxit/errorlog.txt','a')
+    v=[]
+    v.append(dic['oligoid'])
+    v.append(dic['olduser'])
+    v.append(dic['info'])
+    v.append(dic['cdate'])
+    v.append(dic['mdate'])
+    v.append(dic['sequence'])
+    v.append(dic['reference'])
+    v.append(dic['gene_locus'])
+    v.append(dic['organism'])
+    v.append(dic['usage'])
+    v.append(dic['primer_position'])
+    v.append(dic['primer_partner'])
+    v.append(dic['company'])
+    v.append(dic['concentration'])
+    v.append(dic['grade'])
+    failline=''
+    for item in v:
+        failline+='"'
+        failline+=item
+        failline+='",'
+    failline+='"'+message+'"\n'
+    f2.write(failline)
+    f2.close()
+    return failline
+
+def import_oligos(filename):
+    f = open(filename,'r')
+    for line in f:
+        line=unicodeline(line)
+        dic=parseoligo(line)
+        if dic is not None:
+            name = cleanoligoname(dic['oligoid'])
+            if not name:
+                name='unknown'
+            user = cleanuser(dic['olduser'])
+            details = makedetail(dic['oligoid'],
+                                 dic['info'],
+                                 dic['reference'],
+                                 dic['usage'])
+            cdate = cleandate(dic['cdate'])
+            sequence = cleansequence(dic['sequence'])
+            primer_position = cleanpos(dic['primer_position'])
+            primer_partner = cleanoligoname(dic['primer_partner'])
+            usages = getusages(dic['usage'])
```

Comment:
Ignore these for now as they haven't been fully integreated yet but simply copied from old receptor.